compiler/src/dotty/tools/dotc/transform/init/Cache.scala

```diff
@@ -99,7 +105,9 @@ class Cache[Config, Res]:
    */
   def cachedEval(config: Config, expr: Tree, cacheResult: Boolean, default: Res)(eval: Tree => Res): Res =
     this.get(config, expr) match
-    case Some(value) => value
+    case Some(value) =>
+      cacheUsed = true
```

Comment:
For the init checker, if we read from global cache, we can think it's still not used. Maybe move it to the method `get`:

``` Scala
  def get(config: Config, expr: Tree): Option[Res] =
    val res = current.get(config, expr)
    cacheUsed = cacheUsed || res.nonEmpty
    res
```